jobs/generation/PRTestGenerator.groovy

```diff
@@ -0,0 +1,482 @@
+      }
+    }
+  }
+
+  boolean shouldGenerateJob(String scenario, boolean isPR, String architecture, String configuration, String os) {
+    // Generate only PR jobs
+    return isPR;
+  }
+
+  static String getDotnetPath(String branch)
+  {
+    switch (branch) {
+      case 'release/2.0.0':
+      case 'release/2.1':
+      case 'release/2.2':
+        return './Tools/dotnetcli/dotnet'
+      case 'release/3.0':
+      case 'master':
+        return './.dotnet/dotnet'
+      default:
+        assert false : "Unknown branch: '${branch}'"
+        break
+    }
+  }
+
+  static List<String> calculateBuildCommands(def newJob, String scenario, String branch, String architecture,
+                                             String configuration, String os, boolean isBuildOnly) {
+    List<String> buildCommands = []
+    String osGroup = getOSGroup(os)
+    String lowerConfiguration = configuration.toLowerCase()
+
+    // Calculate the build steps, archival, and xunit results
+    switch (os) {
+      case 'Tizen':
+        switch (architecture) {
+          case 'armel':
+            String linuxCodeName = os.toLowerCase()
+
+            assert scenario == 'normal' : "Unsupported scenario: '${scenario}'"
+
+            String xUnitTestBinBase = "./bin/tests/Windows_NT.x64.${configuration}"
+
+            if (!isBuildOnly) {
+              def dockerCommand = Utilities.getDockerCommand(branch)
+
+              // Build tests
+              buildCommands += "${dockerCommand} ./init-tools.sh"
+
+              if (branch == 'release/2.0.0') {
+                buildCommands += "${dockerCommand} ${getDotnetPath(branch)} build ilasm.depproj /p:ProjectDir=/opt/code/ /p:ToolsDir=/opt/code/Tools/"
+              }
+
+              buildCommands += "${dockerCommand} ${getDotnetPath(branch)} restore ./tests/src/Common/test_dependencies/test_dependencies.csproj --packages /opt/code/packages/"
+              buildCommands += """${dockerCommand} bash -c \"for test in \\\$(cat ./tests/testsRunningInsideARM.txt | sed s'/.\$//') ; \\
+              do ${getDotnetPath(branch)} build -c ${configuration} ./tests/src/\\\${test}.*proj \\
+              /p:BuildOS=Linux /p:TargetFrameworkIdentifier=.NETStandard /p:CSharpCoreTargetsPath=Roslyn/Microsoft.CSharp.Core.targets ; done\""""
+
+              buildCommands += "${dockerCommand} chown -R \$(id -u):\$(id -g) ./bin"
+
+              // arm32_ci_test.sh expects that tests are built for Windows_NT
+              buildCommands += "mv ./bin/tests/Linux.x64.${configuration} ${xUnitTestBinBase}"
+
+              // Unpack the corefx binaries
+              String coreFxBinDir = "./bin/CoreFxBinDir"
+              buildCommands += "mkdir ${coreFxBinDir}"
+              buildCommands += "tar -xf ./${getCorefxBinPath(branch)}/build.tar.gz -C ${coreFxBinDir}"
+
+              // Create Core_Root
+              buildCommands += "mkdir -p ${xUnitTestBinBase}/Tests/Core_Root"
+
+              if (branch == 'release/2.0.0') {
```

Comment:
Why do we need separate logic for 2.0?

Reply:
For release/2.0.0 we need to apply patch to build tests on Linux due to it we need to clean up repo as [arm32_ci_script.sh](https://github.com/dotnet/coreclr/blob/b34f2bfee8ca2c5f5508f4d58f5b79dc616dcccc/tests/scripts/arm32_ci_script.sh#L495) expects.

Reply:
How about adding comment about this patch

Reply:
Done